docs/DeveloperGuide.md

```diff
@@ -2,33 +2,249 @@
+1. Introduction
+    - Welcome to DueQuest
+2. Setting Up
+    - Prerequisites
+    - Setting the project
+3. Design
+    - Architecture
+    - Task Component
+    - Command
+    - Managers
+    - Storage
+4. Implementation
+    - Adding
+    - Display 
+    - Storage
+    - Edit
+    - Delete
+5. Documentation
+    - Basic Thing to know
+    - Style guidance
+    - Diagrams
+6. Testing
+    - Running Tests
+    - Types of Tests
+7. Product Scope
+    - Target User Profile
+    - Value Proposition
+    - User Stories
+8. Glossary
 
+## 1. Introduction
+
+### Welcome to DueQuest
+
+Hello! We appreciate your interest in contributing to DueQuest. DueQuest is an application that
+uses a Command Line Interface (CLI) specifically for NUS Students to manage their schedule
+during school. This guide is written to give a clearer understanding for developers who are
+interested in improving the design and implementation of DueQuest.
+
+
+## 2.Setting Up
+
+### Prerequisites
+- JDK 11
+- IntelliJ IDEA
+### Setting the project
+
+1. Fork the repo and clone it into your computer
+2. Configure the JDK in IntelliJ to ensure it is configured to use JDK11.
+3. Click `Import Project` (or `Open or Import` in newer version of Intellij).
+4. IntelliJ IDEA by default has the Gradle plugin installed. If you have disabled it, go
+to `File → Settings → Plugins` to re-enable them.
+5. Locate the build.gradle file (not the root folder as you would do in a normal
+importing) and select it. Click `OK` . If asked, choose to `Open as Project` (not `Open as
+File` ).
+6. Click `OK` to accept the default settings but do ensure that the selected version of
+`Gradle JVM` matches the JDK being used for the project.
+7. Wait for the importing process to finish (could take a few minutes).
+
+## 3. Design
+
+### Architecture
+The diagram below shows a flow chart which gives an overall general picture of how the
+application works whenever the application receives an input from the user and a shows how
+each component interacts with one another for different scenarios. The logic behind the
+application is mostly handled by the `Parser` which converts inputs into various executable
+`Command`.
+
+![](Images/3.1Archi.jpg)
+
+### Task Component
+Each `Task` can be a `Lesson`, `Event` or `Deadline`. Below is a UML diagram showing
+some of the properties and methods that these classes have. A `Task` will be created
+whenever a User wants to add a Task into the `ScheduleManager` or `ModuleManager`.
+
+![](Images/3.2Task.JPG)
+
+### Command
+The logic of what should be executed whenever the application receives an input will be
+handled by parsing these inputs into Commands using a Parser . Below is a UML Diagram
+showing how the `Parser` classes and the `Command` classes interact. Unless the `Command` is
+an `ExitCommand` or `HelpCommand` , the `Command` will be executed to perform an action to
+either the `ScheduleManager` or `ModuleManager` or both.
+
+![](Images/3.3Command.JPG)
```

Comment:
With the composite symbol (the filled diamond), the multiplicity is not really necessary.